src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -691,6 +691,10 @@ private[snowflake] class SnowflakeSQLStatement(
 
   private val log = LoggerFactory.getLogger(getClass)
 
+  private var lastQueryID: String = _
```

Comment:
This should be a thread local, no ?

Reply:
This is only set on failure. I think it would be useful to always set it.

Reply:
yes. It has set up for both failure and success. For example,
      val rs = DefaultJDBCWrapper.executePreparedQueryInterruptibly(statement)
      lastQueryID = rs.asInstanceOf[SnowflakeResultSet].getQueryID
      rs

Reply:
SnowflakeSQLStatement is defined as "class SnowflakeSQLStatement", not "object SnowflakeSQLStatement". The caller doesn't share SnowflakeSQLStatement instance between thread. So it is not necessary to make it thread local.